invenio_rdm_records/records/dumpers/access.py

```diff
@@ -31,10 +31,14 @@ def __init__(self, target_field):
 
     def dump(self, record, data):
         """Dump the grant tokens to the data dictionary."""
-        if record.access:
```

Comment:
ever since the `record.access` field gets initialized even when there's no `record["access"]` specified, this check was insufficient.
in case that there was a way to have `record["access"]` be different from `data["access"]`, this check would've been wrong even then :thinking: